psi-java-bundle-jar/build.gradle.kts

```diff
@@ -70,6 +78,60 @@ tasks.publish {
 }
 
 tasks.shadowJar {
+    dependencies {
+        excludeExternalLibraries()
+        excludeSpineLibraries()
+    }
+    excludeFiles()
+    setZip64(true)  /* The archive has way too many items. So using the Zip64 mode. */
+    archiveClassifier.set("")  /** To prevent Gradle setting something like `osx-x86_64`. */
+    mergeServiceFiles("desc.ref")
+    mergeServiceFiles("META-INF/services/io.spine.option.OptionsProvider")
+}
+
+/**
+ * Exclude libraries we use.
+ *
+ * PSI may also use these libraries, but we want to force their versions.
+ */
+private fun DependencyFilter.excludeExternalLibraries() {
+    ErrorProne.annotations.forEach {
+        exclude(dependency(it))
+    }
+    exclude(dependency(J2ObjC.annotations))
+    exclude(dependency(Guava.lib))
+    exclude(dependency(Gson.lib))
+    Protobuf.libs.forEach {
+        exclude(dependency(it))
+    }
+}
+
+/**
+ * Exclude all Spine libraries from the PSI fat JAR.
+ *
+ * These libraries will be available as separate dependencies.
+ */
+private fun DependencyFilter.excludeSpineLibraries() {
```

Comment:
Let's also discuss. If we use these for the compilation, then technically they should be just regular dependencies for the published artifact.